lib/cubit/security/security_cubit.dart

```diff
@@ -153,7 +166,12 @@ class SecurityCubit extends Cubit<SecurityState> with HydratedMixin<SecurityStat
   @override
   SecurityState? fromJson(Map<String, dynamic> json) {
     final SecurityState state = SecurityState.fromJson(json);
-    _setLockState(state.pinStatus == PinStatus.enabled ? LockState.locked : LockState.unlocked);
+    // Only lock if PIN is enabled and not already in another lock state
+    if (state.pinStatus == PinStatus.enabled && state.lockState != LockState.unlocked) {
```

Comment:
Should we instead check if lockState is initial? Seems like it achieves the same but I find it more readable.

```suggestion
    if (state.pinStatus == PinStatus.enabled && state.lockState == LockState.initial) {
```

Reply:
I have actually reverted to original approach 59dbafedcb1105cd4fa75eba1f920e7b5994a3b0